client/db/src/lib.rs

```diff
@@ -1985,6 +2187,7 @@ impl<Block: BlockT> sc_client_api::backend::Backend<Block> for Backend<Block> {
 				.state_db
 				.reset(state_meta_db)
 				.map_err(sp_blockchain::Error::from_state_db)?;
+			self.blockchain.clear_pinning_cache();
```

Comment:
This is the error branch of the commit operation. We reset the in-memory parts of the state-db to the journaled status from the db. 
Same should be done for the cache, since we never want to return items that are still in db, but only pruned ones.

Reply:
Ops. I didn't saw that it was the error branch 😴